application/F3DConfigFileTools.cxx

```diff
@@ -74,23 +74,27 @@ fs::path F3DConfigFileTools::GetBinaryResourceDirectory()
 }
 
 //----------------------------------------------------------------------------
-fs::path F3DConfigFileTools::GetConfigPath(const std::string& configSearch)
+std::vector<fs::path> F3DConfigFileTools::GetConfigPaths(const std::string& configSearch)
 {
+  std::vector<std::filesystem::path> paths;
+
   fs::path configPath;
-  try
-  {
-    std::vector<fs::path> dirsToCheck;
-    dirsToCheck.emplace_back(F3DConfigFileTools::GetUserConfigFileDirectory());
+  std::vector<fs::path> dirsToCheck = {
+
 #ifdef __APPLE__
-    dirsToCheck.emplace_back("/usr/local/etc/f3d");
+    "/usr/local/etc/f3d",
 #endif
 #ifdef __linux__
-    dirsToCheck.emplace_back("/etc/f3d");
-    dirsToCheck.emplace_back("/usr/share/f3d/configs");
+    "/etc/f3d",
+    "/usr/share/f3d/configs",
 #endif
-    dirsToCheck.emplace_back(F3DConfigFileTools::GetBinaryResourceDirectory() / "configs");
+    F3DConfigFileTools::GetBinaryResourceDirectory() / "configs",
+    F3DConfigFileTools::GetUserConfigFileDirectory(),
```

Comment:
I wonder if we should provide some feedbacks about the directories that have been checked ?

Reply:
Could be done, maybe that should be another PR that also cleans up the search logic? (currently is seems to be split between `F3DConfigFileTools::GetConfigPath` and the top of `ConfigurationOptions::InitializeDictionaryFromConfigFile`)

Reply:
yes, I agree.